js/interactive-guides/microprofile-config/microprofile-config-callback.js

```diff
@@ -264,6 +265,88 @@ var microprofileConfigCallBack = (function() {
         }
     };
 
+    var __listenToBrowserForInjectDefaultConfig = function(webBrowser) {
+        var setBrowserContent = function(currentURL) {
+            webBrowser.setBrowserContent("/guides/iguide-microprofile-config/html/interactive-guides/microprofile-config/download-from-injection.html");
+            contentManager.markCurrentInstructionComplete(webBrowser.getStepName());
+        }
+        
+        webBrowser.contentRootElement.addClass("hidden");
+        webBrowser.addUpdatedURLListener(setBrowserContent);
+    };
+
+    var __listenToBrowserForInjectConfig = function(webBrowser) {
+        var setBrowserContent = function(currentURL) {
+            webBrowser.setBrowserContent("/guides/iguide-microprofile-config/html/interactive-guides/microprofile-config/download-deployment-exception.html");
+            contentManager.markCurrentInstructionComplete(webBrowser.getStepName());
+        }
+        
+        webBrowser.contentRootElement.addClass("hidden");
+        webBrowser.addUpdatedURLListener(setBrowserContent);
+    };
+
+    var __listenToEditorForInjectConfig = function(editor) {
+        var __showWebBrowser = function() {
+            var stepName = editor.getStepName();
+            var content = contentManager.getEditorContents(stepName);
+            if (__checkInjectionEditorContent(content)) {
+                editor.closeEditorErrorBox(stepName);
+                contentManager.showBrowser(stepName, 0);
+                contentManager.addRightSlideClassToBrowser(stepName, 0);
+                contentManager.markCurrentInstructionComplete(stepName);
+                //contentManager.updateWithNewInstructionNoMarkComplete(stepName);
+            } else {
+                // display error and provide link to fix it
+                editor.createErrorLinkForCallBack(stepName, true, __addInjectConfigToEditor);
+            }
+        };
+        editor.addSaveListener(__showWebBrowser);
+    };
+
+    var __isInjectParamInAnnotation = function(annotationParams) {
+        var allMatch = false; 
+        if (annotationParams.length === 1) {
+            var param1 = annotationParams[0];
+                  
+            if (param1 === "name=\"download-url\"") {
```

Comment:
Change it to download_url as I've changed it following the best practice to use _ rather than -.